roles/openshift_facts/tasks/main.yml

```diff
@@ -64,8 +64,7 @@
     role: common
     local_facts:
       debug_level: "{{ openshift_debug_level | default(2) }}"
-      # TODO: Deprecate deployment_type in favor of openshift_deployment_type
-      deployment_type: "{{ openshift_deployment_type | default(deployment_type) }}"
+      deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
```

Comment:
This will fail if deployment_type is not set...  you will need to nest the second default filter similar to my other comment.